src/DistributedComputations/distributed_fft_tridiagonal_solver.jl

```diff
@@ -0,0 +1,333 @@
+                   global_grid :: G
+                    local_grid :: L
+    batched_tridiagonal_solver :: B
+                   source_term :: R
+                       storage :: S
+                        buffer :: β
+end
+
+architecture(solver::DistributedFourierTridiagonalPoissonSolver) =
+    architecture(solver.global_grid)
+
+"""
+    DistributedFourierTridiagonalPoissonSolver(global_grid, local_grid)
+
+Return an FFT-based solver for the Poisson equation evaluated on a `local_grid`
+with one stretched direction
+
+```math
+∇²φ = b
+```
+
+for `Distributed` architectures.
+
+Supported configurations
+========================
+
+In the following, `Nx`, `Ny`, and `Nz` are the number of grid points of the **global** grid, 
+in the `x`, `y`, and `z` directions, while `Rx`, `Ry`, and `Rz` are the number of ranks in the
+`x`, `y`, and `z` directions, respectively. Furthermore, 'pencil' decomposition refers to a domain 
+decomposed in two different directions (i.e., with `Rx != 1` and `Ry != 1`), while 'slab' decomposition 
+refers to a domain decomposed only in one direction, (i.e., with either `Rx == 1` or `Ry == 1`).
+Additionally, `storage` indicates the `TransposableField` used for storing intermediate results;
+see [`TransposableField`](@ref).
+
+1. Three dimensional configurations with 'pencil' decompositions in ``(x, y)`` 
+where `Ny ≥ Rx` and `Ny % Rx = 0`, and `Nz ≥ Ry` and `Nz % Ry = 0`.
+
+2. Two dimensional configurations decomposed in ``x`` where `Ny ≥ Rx` and `Ny % Rx = 0`
+    
+!!! warning "Unsupported decompositions"
+    _Any_ configuration decomposed in ``z`` direction is _not_ supported.
+    Furthermore, any ``(x, y)`` decompositions other than the configurations mentioned above are also _not_ supported.
+    
+Algorithm for pencil decompositions
+============================================
+
+For pencil decompositions (useful for three-dimensional problems),
+there are two forward transforms, two backward transforms, one tri-diagonal matrix inversion
+and a variable number of transpositions that require MPI communication, dependent on the 
+strethed direction:
+
+- a stretching in the x-direction requires four transpositions
+- a stretching in the y-direction requires six transpositions
+- a stretching in the z-direction requires eight transpositions
+
+!!! warning "Computational cost"
+    Because of the additional transpositions, a stretching in the x-direction
+    is computationally cheaper than a stretching in the y-direction, and the latter
+    is cheaper than a stretching in the z-direction
+
+In our implementation we require `Nz ≥ Ry` and `Nx ≥ Ry` with the additional constraint 
+that `Nz % Ry = 0` and `Ny % Rx = 0`.
+
+X - stretched algorithm
+========================
+
+1. `storage.zfield`, partitioned over ``(x, y)`` is initialized with the `rhs`.
+2. Transform along ``z``.
+3. Transpose + communicate to `storage.yfield` partitioned into `(Rx, Ry)` processes in ``(x, z)``.
```

Comment:
Could you state explicitly right after the word transpose as to what fields are being transposed?  I think it would make it easier to read.